### PR TITLE
use ImplicitZero instead of Return unit

### PR DESCRIPTION
**Pull request by mexx**
Motivation: fsprojects/FSharp.Control.AsyncSeq#38

[UserVoice](https://fslang.uservoice.com/forums/245727-f-language/suggestions/10999326-interpret-do-action-as-let-action-in-zer)

This PR don't have any new test, any guidance where to find the tests for CEs and to put new for the change is welcome.

### CODE REVIEW

**Comment by KevinRansom** [comment]
Perhaps here:  tests\fsharp\core\comprehensions\test.fsx  ?

**Comment by mexx** [comment]
I've added the tests. Comments welcome.

File: tests/fsharp/core/comprehensions/test.fsx [hunk 1]
```diff
@@ -1475,6 +1475,49 @@ module EnumPatternWithFunkyTypes_FSharp_1_0_13904 =
...
+        member b.Combine(x:list<'a>,y:list<'a>) :list<'a> = List.append x y
+        member b.Delay(f:unit->list<'a>) :list<'a> = f()
+        member b.Zero() :list<'a> = onZero(); []
+        member b.Yield(x:'a) :list<'a> = [x]
+
+    let sideEffectListWithZero onZero = SideEffectListWithZeroBuilder(onZero)
+
+    module SideEffectListTests =
+        #if Portable
+        let printfn s = printfn "%s" s
```
Comment (enricosada) [inline tests/fsharp/core/comprehensions/test.fsx]: I know you copy/paste from near tests (good), but maybe this is good opportunity to cleanup this duplication, adding the `printfn` function at top level

File: tests/fsharp/core/comprehensions/test.fsx [hunk 2]
```diff
@@ -1475,6 +1475,49 @@ module EnumPatternWithFunkyTypes_FSharp_1_0_13904 =
...
+        #endif
+
+        let x0a : list<int> * int * int =
+            let calledReturn = ref 0
+            let onReturn () = calledReturn := !calledReturn + 1
```
Comment (enricosada) [inline tests/fsharp/core/comprehensions/test.fsx]: the use of `ref` instead of `mutable` ( f# 4) here it's beacuse this need to run as f# 3.1 test?

**Comment by dsyme** [comment]
I've marked the language suggestion as approved https://fslang.uservoice.com/forums/245727-f-language/suggestions/10999326-interpret-do-action-as-let-action-in-zer

File: tests/fsharp/core/comprehensions/test.fsx (hunk 1 above)
Comment (mexx) [inline tests/fsharp/core/comprehensions/test.fsx]: @enricosada maybe, or better to get rid of the root cause. I mean why it isn't simply working for `Portable` too?

**Comment by dsyme** [comment]
Thanks for the contribution @mexx !!

**Pull request merged**